src/notification.c

```diff
@@ -584,6 +717,11 @@ parse_notification (GVariantBuilder  *builder,
 
           g_variant_builder_add (builder, "{sv}", key, value);
         }
+      else if (strcmp (key, "markup-body") == 0)
```

Comment:
I think it might be worth it to validate that all the strings here are valid utf-8.

Reply:
`GVariant` does that already. UTF-8 validation is expensive and should only be done on data that crosses a process boundary (i.e. stuff sent over IPC, or received from the network, or loaded from disk). D-Bus does UTF-8 validation for string types already, as does `GVariant`. Could the data here come from some other source?

Reply:
Alright, all good then.